### DISCUSSION TITLE
Template can't evaluate field in type *models.Record

### DISCUSSION THREAD

**Discussion by f11xter** [pocketbase/pocketbase#4722] [Q&A] [answered]
Given a `posts` collection with a `text` field, the following range template:
```js
// main.pb.js
const posts = $app.dao().findRecordsByFilter("posts",
  `text != ""`,
  "-created",
  10,
  0,
);

const html = $template
  .loadFiles(`${__hooks}/index.html`)
  .render({ posts })

// index.html
{{range .posts}} {{.Created}} {{.Text}} {{end}}
```
throws
```
template: index.html:13:40: executing "index.html" at <.Text>: can't evaluate field Text in type *models.Record
```
https://github.com/pocketbase/pocketbase/discussions/3419 suggested changing the property names to pascal case which worked for `Created` but not for `Text`.

How should I solve this?

And a secondary question: is there a way to pass an empty filter to findRecordsByFilter (instead of `text != ""`)?

**Comment by ganigeorgiev** [comment, marked as answer]
Dynamic record schema fields must be accessed through the designated [getters](https://pocketbase.io/docs/go-records/#getset-record-fields), so you can try with `.Get 'text'` (_with the [refactoring](https://github.com/pocketbase/pocketbase/discussions/4355) all fields, including the system one, will be part of the collections schema and their access will be normalized_).

>  is there a way to pass an empty filter to findRecordsByFilter (instead of `text != ""`)?

You can try with a simpler dummy filter that will always evaluate to true and has very low cost, eg. `1=1` (_I'll consider removing the method non-empty filter requirement in the future_). Alternatively if you just want to apply a limit to the collections record retrieval you can try with a custom query like:

```js
const records = arrayOf(new Record);
$app.dao().recordQuery("posts").limit(10).orderBy("created DESC").all(records)
```

**Reply by f11xter** [reply]
Ah I missed the getters section. Thank you!
